pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -844,10 +844,16 @@ func buildRouteVHostDomains(svc *model.Service) []string {
 	return domains
 }
 
-func buildWaypointInboundHTTPRouteConfig(lb *ListenerBuilder, svc *model.Service, cc inboundChainConfig) *route.RouteConfiguration {
+func buildWaypointInboundHTTPRouteConfig(lb *ListenerBuilder, svc *model.Service, cc inboundChainConfig) (out *route.RouteConfiguration) {
+	defer func() {
+		efw := lb.push.EnvoyFilters(lb.node, svc)
```

Comment:
fwiw by fetching efw here we need to pull this once per service-port when we _could_ do it once per service. not a huge deal though